### PR TITLE
README.md: update readme

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve?

Problem Summary: update readme to be more user friendly.

### What is changed and how it works?

- adjust the sections
- add a TOC
- add data migration

### Check List 

Tests 

- No code

## Release Note

- N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,44 +10,68 @@
...
 
-- __MySQL Compatible Syntax__
+- [English](https://pingcap.com/docs)
+- [简体中文](https://pingcap.com/docs-cn)
 
-    TiDB acts like it is a MySQL 5.7 server to your applications. You can continue to use all of the existing MySQL client libraries, and in many cases, you will not need to change a single line of code in your application. Because TiDB is built from scratch, not a MySQL fork, please check out the list of [known compatibility differences](https://pingcap.com/docs/v3.0/reference/mysql-compatibility/).
+## Quick start
 
-- __Distributed Transactions with Strong Consistency__
+- [Quick start guide](https://docs.pingcap.com/zh/tidb/v4.0/quick-start-with-tidb)
```
Comment (siddontang) [inline README.md:42]: em, don't link the Chinese doc, or maybe we can use two versions.

Comment (queenypingcap) [reply README.md:42]: Please add two versions here.

File: README.md [hunk 2]
```diff
@@ -10,44 +10,68 @@
...
+* [Acknowledgments](#acknowledgments)
+
+<!-- vim-markdown-toc -->
 
 ## What is TiDB?
 
 TiDB ("Ti" stands for Titanium) is an open-source NewSQL database that supports Hybrid Transactional and Analytical Processing (HTAP) workloads. It is MySQL compatible and features horizontal scalability, strong consistency, and high availability.
 
-- __Horizontal Scalability__
+For a detailed introduction to TiDB architecture, see [TiDB Architecture](https://pingcap.com/docs/dev/architecture/).
```
Comment (queenypingcap) [inline README.md:33]: I recommend that we keep the architecture diagram here so users won't have to jump through different links to find out.

File: README.md [hunk 3]
```diff
@@ -10,44 +10,68 @@
...
 
-    TiDB expands both SQL processing and storage by simply adding new nodes. This makes infrastructure capacity planning both easier and more cost-effective than traditional relational databases which only scale vertically.
+## Documentation
 
-- __MySQL Compatible Syntax__
+- [English](https://pingcap.com/docs)
+- [简体中文](https://pingcap.com/docs-cn)
 
-    TiDB acts like it is a MySQL 5.7 server to your applications. You can continue to use all of the existing MySQL client libraries, and in many cases, you will not need to change a single line of code in your application. Because TiDB is built from scratch, not a MySQL fork, please check out the list of [known compatibility differences](https://pingcap.com/docs/v3.0/reference/mysql-compatibility/).
+## Quick start
```
Comment (queenypingcap) [inline README.md:40]: Please add links to the English content only.

File: README.md [hunk 4]
```diff
@@ -10,44 +10,68 @@
...
+* [Roadmap](#roadmap)
+* [Contributing](#contributing)
+* [License](#license)
+* [Acknowledgments](#acknowledgments)
+
+<!-- vim-markdown-toc -->
 
 ## What is TiDB?
 
 TiDB ("Ti" stands for Titanium) is an open-source NewSQL database that supports Hybrid Transactional and Analytical Processing (HTAP) workloads. It is MySQL compatible and features horizontal scalability, strong consistency, and high availability.
```
Comment (queenypingcap) [inline README.md:32]: Please keep the features and the introductions so users can know why they should choose TiDB at first sight.

**Pull request closed without merging**